core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -659,6 +659,14 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     setLocalProperty(SparkContext.SPARK_JOB_INTERRUPT_ON_CANCEL, null)
   }
 
+  /**
+   * Execute a block of code in a scope.
```

Comment:
Also, I wonder if we should call this "operator scope" or something since it's tied to user-facing operators.